airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -40,6 +53,25 @@ public StandardSyncOutput run(final JobRunConfig jobRunConfig,
                                 final StandardSyncInput syncInput,
                                 final UUID connectionId) {
 
+    final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput()
+        .withConnectionConfiguration(syncInput.getSourceConfiguration());
+    final StandardCheckConnectionInput destinationConfiguration =
+        new StandardCheckConnectionInput().withConnectionConfiguration(syncInput.getDestinationConfiguration());
+
+    final StandardCheckConnectionOutput sourceCheckResponse = checkActivity.check(jobRunConfig, sourceLauncherConfig, sourceConfiguration);
+    if (sourceCheckResponse.getStatus() == Status.FAILED) {
+      final Exception ex = new IllegalArgumentException(sourceCheckResponse.toString());
+      final FailureReason checkFailureReason = FailureHelper.checkFailure(ex, Long.valueOf(jobRunConfig.getJobId()), Math.toIntExact(jobRunConfig.getAttemptId()), FailureReason.FailureOrigin.SOURCE);
+      return new StandardSyncOutput().withFailures(List.of(checkFailureReason));
+    }
```

Comment:
Here's what I *think* is happening here.

1. We run the check for the Source
2. These workflow activities never throw, so we check the response status to see if things have worked
3. If there's a problem (`sourceCheckResponse.getStatus() == Status.FAILED`), we want to exit the workflow early.  Returning the same response payload, a `StandardSyncOutput` seems like the right thing do.  If the StandardSyncOutput contains a failure reason structured properly, the sync should not retry.